efi/secureboot_policy.go

```diff
@@ -198,14 +192,17 @@ func (s *shimImageHandle) version() (shimVersion, error) {
 	}
 	for scanner.Scan() {
 		m := shimIdentVersionRE.FindStringSubmatch(scanner.Text())
-		if len(m) == 3 {
+		if len(m) > 1 {
 			major, err := strconv.ParseUint(m[1], 10, 0)
 			if err != nil {
 				return shimVersion{}, fmt.Errorf("invalid major version \"%s\"", m[1])
 			}
-			minor, err := strconv.ParseUint(m[2], 10, 0)
-			if err != nil {
-				return shimVersion{}, fmt.Errorf("invalid minor version \"%s\"", m[2])
+			var minor uint64
+			if len(m) > 2 && len(m[2]) > 0 {
```

Comment:
and this as len(m) == 3 as we know how many capturing groups we have?

Reply:
Looking into more detail how `FindStringSubmatch` works, it seems that either it returns nil or it is successful and it returns an array with 1 + number of groups elements (there needs to be a way to know which group is an element of the array for), so the checks on len(m) are not needed, we just need to check `len(m[2])` here. But probably not worth changing this now as this code will not be used in master afaik.